hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/utils/ConversionUtils.java

```diff
@@ -721,15 +745,19 @@ private static com.hedera.pbj.runtime.io.buffer.Bytes bloomFor(@NonNull final Lo
     }
 
     private static Address longZeroAddressIn(@NonNull final MessageFrame frame, @NonNull final Address address) {
-        return isLongZero(address)
+        final var shard = ((HederaWorldUpdater) frame.getWorldUpdater()).getShardNum();
+        final var realm = ((HederaWorldUpdater) frame.getWorldUpdater()).getRealmNum();
+        return isLongZero(shard, realm, address)
                 ? address
                 : asLongZeroAddress(
```

Comment:
bug `asLongZeroAddress` calling `asEvmAddress` is not adding in the shard/realm.  also affects `explicitAddressOf` in this class.